test/browser/async_exports.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <assert.h>
+
+#include <emscripten.h>
+
+enum UnsuspendFunctionId {
+    UNSUSPEND_EXPORT = 0,
+    OTHER_UNSUSPEND_EXPORT,
+};
+
+EM_ASYNC_JS(int, suspend, (int value, int functionId), {
+  return await new Promise(resolve => {
+    window.setTimeout(() => {
+        Module['asm'][functionId === 0 ? 'unsuspendExport' : 'otherUnsuspendExport'].apply(null, [value]);
+    }, 0);
+    Module.awakeTest = resolve;
+  });
+});
+
+EM_ASYNC_JS(void, unsuspend, (int value), {
+  Module.awakeTest(value);
+});
+
+extern "C" EMSCRIPTEN_KEEPALIVE void unsuspendExport(int value) {
+    unsuspend(value);
+}
+
+extern "C" EMSCRIPTEN_KEEPALIVE void otherUnsuspendExport(int value) {
+    unsuspend(value);
+}
+
+int main() {
+#ifdef BAD
+  EM_ASM({
+    Module.expectError = true;
+  });
+#endif
+
+  EM_ASM({
+    window.disableErrorReporting = true;
+    // For some reason, e does not contain an error in case of bad export instrumentation.
+    // Send the result manually here.
+    window.onerror = function(e) {
```

Comment:
This seems rather hacky.. i would hope we could find a better way?  But its hard for me to say without debugging the the code why this would be needed?

Reply:
Test replaced